app/src/main/java/cloudcity/CloudCityParamsRepository.java

```diff
@@ -0,0 +1,154 @@
+package cloudcity;
+
+import static cloudcity.CloudCityConstants.CLOUD_CITY_SERVER_URL;
+import static cloudcity.CloudCityConstants.CLOUD_CITY_TOKEN;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import androidx.annotation.Nullable;
+
+public class CloudCityParamsRepository {
+    private static final String SHARED_PREFS_NAME = "cloud_city_prefs";
+
+    private Context context;
+    private static CloudCityParamsRepository instance;
+
+    private SharedPreferences sharedPrefs;
+
+    private String serverUrl;
+    private String serverToken;
+
+
+    private CloudCityParamsRepository(Context context) {
+        this.context = context;
+        this.sharedPrefs = context.getSharedPreferences(SHARED_PREFS_NAME, Context.MODE_PRIVATE);
+    }
+
+    /**
+     * Initializes the repository, loads saved data and prefills it with data if no data was found
+     *
+     * @param ctx the context to use
+     */
+    public static void initialize(Context ctx) {
+        if (instance != null) {
+            throw new IllegalStateException("Already initialized!");
+        } else {
+            // Since instance is null, this branch is already the first part of
+            // the double-locked synchronized singleton creation
+            synchronized (CloudCityParamsRepository.class) {
+                if (instance == null) {
+                    instance = new CloudCityParamsRepository(ctx);
+                }
+            }
+        }
+        instance.loadData();
+        if (!instance.isParamPresent(instance.getServerUrl()) || !instance.isParamPresent(instance.getServerToken())) {
+            instance.prefillWithData();
+        }
+    }
+
+    public static CloudCityParamsRepository getInstance() {
+        if (instance == null) {
+            throw new IllegalStateException("Must call initialize() first!");
+        }
+
+        return instance;
+    }
+
+    private void loadData() {
+        serverUrl = sharedPrefs.getString(CLOUD_CITY_SERVER_URL, "");
+        serverToken = sharedPrefs.getString(CLOUD_CITY_TOKEN, "");
+    }
+
+    private void prefillWithData() {
+        String sharkToken = "252|GI2CTemW2EX3TmQaDmCzYg1xrs3VEDslnGrfCwp245a84b22";
+        String chungaToken = "68|5LGMoNAd0mck4bmMaGdj2GqjqqYUB1NyqtbSrpFB82303173";
+
+        serverUrl = "staging.app.cloudcities.co";
+        serverToken = sharkToken;
+
+        sharedPrefs
+                .edit()
+                .putString(CLOUD_CITY_SERVER_URL, serverUrl)
+                .putString(CLOUD_CITY_TOKEN, serverToken)
+                .commit();
+    }
```

Comment:
it's a non-issue since it's just for the staging server. it *will* become an issue when staging/prod server build variants come into play, but is yet a non-issue. resolving.